File: Front/src/types/notification.ts
export type NotificationType =
  | "StageStatusChanged"
  | "StageAssigned"
  | "NewChatMessage"
  | "DeadlineApproaching"

export interface NotificationDto {
  id: string
  type: NotificationType
  title: string
  message: string
  isRead: boolean
  relatedProjectId: string | null
  relatedStageId: string | null
  createdAt: string
}

File: Back/src/Core/Entities/Notification.cs
using Core.Enums;

namespace Core.Entities;

public class Notification : BaseEntity
{
    public Guid UserId { get; set; }

    public NotificationType Type { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Message { get; set; } = string.Empty;

    public bool IsRead { get; set; }

    public Guid? RelatedProjectId { get; set; }

    public Guid? RelatedStageId { get; set; }
}
